plugins/spi/SPIOutput.cpp

```diff
@@ -180,26 +185,67 @@ SPIOutput::SPIOutput(const UID &uid, SPIBackendInterface *backend,
   m_spi_device_name = FilenameFromPathOrPath(m_backend->DevicePath());
 
   PersonalityCollection::PersonalityList personalities;
-  personalities.push_back(Personality(m_pixel_count * WS2801_SLOTS_PER_PIXEL,
-                                      "WS2801 Individual Control"));
-  personalities.push_back(Personality(WS2801_SLOTS_PER_PIXEL,
-                                      "WS2801 Combined Control"));
-  personalities.push_back(Personality(m_pixel_count * LPD8806_SLOTS_PER_PIXEL,
-                                      "LPD8806 Individual Control"));
-  personalities.push_back(Personality(LPD8806_SLOTS_PER_PIXEL,
-                                      "LPD8806 Combined Control"));
-  personalities.push_back(Personality(m_pixel_count * P9813_SLOTS_PER_PIXEL,
-                                      "P9813 Individual Control"));
-  personalities.push_back(Personality(P9813_SLOTS_PER_PIXEL,
-                                      "P9813 Combined Control"));
-  personalities.push_back(Personality(m_pixel_count * APA102_SLOTS_PER_PIXEL,
-                                      "APA102 Individual Control"));
-  personalities.push_back(Personality(APA102_SLOTS_PER_PIXEL,
-                                      "APA102 Combined Control"));
+  // personality description is max 32 characters
+
+  personalities.insert(personalities.begin() + PERS_WS2801_INDIVIDUAL - 1,
+    Personality(m_pixel_count * WS2801_SLOTS_PER_PIXEL,
```

Comment:
Can we indent this to align with the ( above ideally, or otherwise four in from personalities depending on line length.

Reply:
do you mean like this:
```
  personalities.insert(personalities.begin() + PERS_WS2801_INDIVIDUAL - 1,
                       Personality(m_pixel_count * WS2801_SLOTS_PER_PIXEL,
                                   "WS2801 Individual Control"));
that would fit the line length.
```

Reply:
The top one please, it's the two arguments to insert, then the two to Personality.